x-pack/plugins/lens/public/indexpattern_datasource/indexpattern_suggestions.ts

```diff
@@ -455,6 +455,10 @@ export function getDatasourceSuggestionsFromCurrentState(
           ({ name }) => name === indexPattern.timeFieldName
         );
 
+        const hasNumericDimension =
+          buckets.length === 1 &&
+          buckets.some((columnId) => layer.columns[columnId].dataType === 'number');
```

Comment:
Instead of looking for numeric operations, which can include the number range aggregation too, could you filter for any existing `column.scale === 'interval'`? This should allow the timestamp to be added when it's a "Custom ranges" option, which is currently not added.